### PR TITLE
License Agreement and Documentation added

### PR DESCRIPTION
**Pull request by kaiwalyakoparkar**
#### Summary
Required changes are done in the readme file & Licence file. The year and the owner information was missing from the license agreement so Added that information and also added this license agreement in the readme.md file.

#### Release Note

```release-note
None
```

### CODE REVIEW

**Comment by jasonblais** [comment]
@kaiwalyakoparkar thank you for your interest contributing! However, we won't be able to accept these changes:

1. The Apache 2.0 license itself should be left unmodified per legal advice. Modifying the example text will cause it to not be recognized as Apache 2.0 by a large number of license file identifiers. Instead, we include the appropriate copyright year and author (Mattermost) at the top of each source file in the repository. You can see comparable discussion in the Kubernetes repository: https://github.com/kubernetes/kubernetes/pull/65 https://github.com/kubernetes/kubernetes/pull/37305
2. Including part of the LICENSE file in the README.md may be confusing and omit critical information that conflicts with the license file itself. Thus, we prefer to simply refer users to the LICENSE file for the exact license text.

Let me know if you have any questions, and thanks for taking the time to contribute to Mattermost.

**Comment by kaiwalyakoparkar** [comment]
@jasonblais please check now, I have updated my PR with some changes. Thanks for letting me know the rules of the apache license.

**Comment by jasonblais** [comment]
Thank you @kaiwalyakoparkar!

GitHub did recently make it easier to find the license file for a repository, so I'm leaning towards not adding a link to the README since it would be repetitive.

image

What do you think?

**Comment by kaiwalyakoparkar** [comment]
Thankyou @jasonblais ,
According to me whenever a person goes through the readme file that time he does not pay attention to the side pane. Also when we scroll down to read the readme file that time this feature hides/scrolls up. That's the reason I want to add this change so that the reader/ user gets it in the documentation part itself and does not have to search for it separately. Also many times they forget to read it and it may cause a blunder. I hope my thoughts are clear. What do you think?

File: README.md [hunk 1]
```diff
@@ -73,6 +73,10 @@ Receive notifications of critical security updates. The sophistication of online
 - [See who's using Mattermost](https://mattermost.com/blog/category/customer-stories/)
 - [Browse over 700 Mattermost integrations](https://integrations.mattermost.com/)
 
+## License
+
+To read the license agreement visit [here](https://github.com/mattermost/mattermost-server/blob/master/LICENSE.txt)
```
Comment (jasonblais) [inline README.md:78]:
```suggestion
See the [LICENSE file](https://github.com/mattermost/mattermost-server/blob/master/LICENSE.txt) for license rights and limitations.
```

**Review by jasonblais — CHANGES_REQUESTED** [review]
@kaiwalyakoparkar I'm good with that use case, thank you! I'm proposing a small rewording, given "license agreement" may refer to the Enterprise Edition License agreement of the commercial version (https://mattermost.com/pricing/).

File: README.md [hunk 2]
```diff
@@ -73,6 +73,10 @@ Receive notifications of critical security updates. The sophistication of online
 - [See who's using Mattermost](https://mattermost.com/blog/category/customer-stories/)
 - [Browse over 700 Mattermost integrations](https://integrations.mattermost.com/)
 
+## License
+
+See the [LICENSE file](LICENSE.txt) for license rights and limitations.
```
Comment (kaiwalyakoparkar) [inline README.md:78]: @jasonblais I have changed the text as asked by you also I have shortened the URL to the license agreement file

Comment (kaiwalyakoparkar) [reply README.md:78]: @jasonblais please let me know if the shortened URL will work or I may change it to a full link

**Review by jasonblais — APPROVED** [review]
Looks good to me, thank you!

**Review by jespino — APPROVED** [review]

**Review by wiersgallak — APPROVED** [review]

**Comment by jasonblais** [comment]
Thanks @kaiwalyakoparkar for the contribution!

**Pull request merged**